File: voice-telephony/ringout/make-ringout-call/code-samples/createRingOutCall.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$accountId = '<ENTER VALUE>';
$extensionId = '<ENTER VALUE>';

// POST BODY
$body = array(
    'from' => array(
        'phoneNumber' => '<ENTER VALUE>',
        'forwardingNumberId' => '<ENTER VALUE>'
    ),
    'to' => array(
        'phoneNumber' => '<ENTER VALUE>'
    ),
    'callerId' => array(
        'phoneNumber' => '<ENTER VALUE>'
    ),
    'playPrompt' => true,
    'country' => array(
        'id' => '<ENTER VALUE>'
    )
)

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));
$r = $platform->post("/restapi/v1.0/account/{$accountId}/extension/{$extensionId}/ring-out", $body);
?>

File: voice-telephony/ringout/make-ringout-call/code-samples/createRingOutCall.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";
string extensionId = "<ENTER VALUE>";

// POST BODY
MakeRingOutRequest makeRingOutRequest = new MakeRingOutRequest {
    from = new MakeRingOutCallerInfoRequestFrom {
        phoneNumber = "<ENTER VALUE>",
        forwardingNumberId = "<ENTER VALUE>"
    },
    to = new MakeRingOutCallerInfoRequestTo {
        phoneNumber = "<ENTER VALUE>"
    },
    callerId = new MakeRingOutCallerInfoRequestTo {
        phoneNumber = "<ENTER VALUE>"
    },
    playPrompt = true,
    country = new MakeRingOutCoutryInfo {
        id = "<ENTER VALUE>"
    }
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).Extension(extensionId).RingOut().Post(makeRingOutRequest);
// PROCESS RESPONSE
